keps/sig-node/4818-allow-zero-value-for-sleep-action-of-prestop-hook/README.md

```diff
@@ -0,0 +1,580 @@
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+N/A
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+Basic functionality
+
+- Create a simple pod with a container that runs a long-running process.
+- Add a preStop hook to the container configuration, using the new sleepAction with a sleep duration of `0`.
+- Delete the pod and observe the time it takes for the container to terminate.
+- Verify that the container terminates immediately without sleeping.
+
+### Graduation Criteria
+
+#### Alpha
+- Feature implemented behind a feature flag
+- Initial unit/e2e tests completed and enabled
+
+#### Beta
+- Gather feedback from developers and surveys
+- Additional e2e tests are completed (if needed)
+- No trouble reports from alpha release
+
+#### GA
+- No trouble reports with the beta release, plus some anecdotal evidence of it being used successfully.
+
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+
+The previous PreStop Sleep Action behavior will not be broken. Users can continue to use their hooks as it is. To use this enhancement, users need to enable the feature gate, and set the sleep duration as zero in their prestop hook’s sleep action.
+
+#### Downgrade
+
+If the kube-apiserver is downgraded to a version where the feature gate is not supported (<v1.32), no new resources can be created with a PreStop sleep duration of zero seconds. Existing resources will continue to function, but they cannot be updated since the validation for the sleep duration would fail if the value is set to zero. Cluster admins can update their resources to have a non-zero value for the sleep duration and migrate their resources if they want to update it.
+
+If the feature gate is turned off after being enabled, no new resources can be created with PreStop sleep duration of zero seconds. Existing resources will continue to function, but as discussed in the Proposal section, the sleep duration will need to be updated to a non-zero value if the resource has to be updated.
```

Comment:
This is outdated vs proposal section - existing resources should continue to work as described in the proposal (in particular this one: "Existing resources with a PreStop sleep duration set to zero will continue to run and use a sleep duration of zero seconds. These can be updated and the zero sleep duration would continue to work.")

Plese update